src/auto-complete.directive.ts

```diff
@@ -129,8 +131,11 @@ export class NguiAutoCompleteDirective implements OnInit {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes['ngModel']) {
-      this.ngModel = this.setToStringFunction(changes['ngModel'].currentValue);
+    // if (changes['ngModel']) {
+    //   this.ngModel = this.setToStringFunction(changes['ngModel'].currentValue);
+    // }
+    if (changes["source"]) {
+      this.sourceChanges.emit(this.source);
```

Comment:
Actually, if class doesn't implement lifecycle hook interface, it doesn't mean that this method doesn't work. Do not forget that interfaces exists only at TS code, so at runtime Angular will call this method as expected.